api/envoy/config/core/v3/http_service.proto

```diff
@@ -0,0 +1,48 @@
+syntax = "proto3";
+
+package envoy.config.core.v3;
+
+import "envoy/config/core/v3/base.proto";
+
+import "google/protobuf/duration.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.config.core.v3";
+option java_outer_classname = "HttpServiceProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/config/core/v3;corev3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: HTTP services]
+
+// HTTP service configuration.
+// [#next-free-field: 6]
+message HttpService {
+  // The name of the upstream HTTP cluster. SSL credentials will be supplied
+  // in the :ref:`Cluster <envoy_v3_api_msg_config.cluster.v3.Cluster>` :ref:`transport_socket
+  // <envoy_v3_api_field_config.cluster.v3.Cluster.transport_socket>`.
+  string cluster_name = 1 [(validate.rules).string = {min_len: 1}];
+
+  // The path to use for the service.
+  // The default is an empty string.
+  string path = 2;
+
+  // Specifies a list of HTTP headers that should be added to each request
+  // handled by this virtual host.
+  repeated HeaderValueOption request_headers_to_add = 3
```

Comment:
Should this be a more general `HeaderMutation`?

Reply:
I tried following @adisuissa's comment above:

>I suggest following the naming conventions (and comments) as in [main/api/envoy/config/route/v3/route_components.proto](https://github.com/envoyproxy/envoy/blob/main/api/envoy/config/route/v3/route_components.proto?rgh-link-date=2023-09-12T14%3A57%3A01Z), to keep consistency between the two. Also note that custom-headers can be both on the request path and maybe on the response path (no need to add response-path support, just to clarify that this is only for requests).

`route_components.proto` uses `HeaderValueOption` so I thought to use the same.